packages/block-editor/src/store/reducer.js

```diff
@@ -1416,9 +1416,19 @@ export function selection( state = {}, action ) {
 			}
 	}
 
+	const selectionStart = selectionHelper( state.selectionStart, action );
+	const selectionEnd = selectionHelper( state.selectionEnd, action );
+
+	if (
+		selectionStart === state.selectionStart &&
+		selectionEnd === state.selectionEnd
+	) {
+		return state;
```

Comment:
I also noticed that we are continuously creating a new object here. I'm curiously if this improves performance.